lib/Parse/ParseType.cpp

```diff
@@ -1574,7 +1588,7 @@ bool Parser::canParseType() {
       return false;
     break;
   case tok::oper_prefix:
-    if (Tok.getText() != "~") {
+    if (Tok.getText() != "~" && Tok.getText() != "-") {
```

Comment:
Similar here, does this mean that we return true from `canParseType` for `-String`?